src/pages/[...slug].tsx

```diff
@@ -56,28 +56,41 @@ export async function getStaticProps(
   const addResourceToCollection = nodeMeta[type]?.additionalNode
 
   /** Check for is collection variable and determines if it is a collection or a single resource */
-
   if (!isCollection) {
     node = await drupalClient.getResourceFromContext<NodeTypes>(path, context, {
-      params: nodeMeta[type].params.getQueryObject(),
+      params: {
+        ...nodeMeta[type].params.getQueryObject(),
+      },
     })
   }
+
   if (isCollection) {
     node = await drupalClient.getResourceCollectionFromContext<NodeTypes>(
       type,
       context,
-      { params: nodeMeta[type].params.getQueryObject() }
+      {
+        params: {  
+          'filter[drupal_internal__nid][value]': path.entity.id,
+          ...nodeMeta[type].params.getQueryObject(),
+        },
+      }
     )
+
     /** Check for is additionalNode variable and returns data for additional collection or single resource */
     if (addResourceToCollection) {
       additionalNode =
         await drupalClient.getResourceCollectionFromContext<NodeTypes>(
           addResourceToCollection,
           context,
-          { params: nodeMeta[type].additionalParams.getQueryObject() }
+          {
+            params: {
+              'filter[field_listing.drupal_internal__nid][value]' : path.entity.id, // Todo make the filter option dynamic
+              ...nodeMeta[type].additionalParams.getQueryObject(),
+            },
```

Comment:
Adding the filter here based on `path.entity.id` fixes the issue, however this needs to be made more dynamic